src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -142,6 +145,12 @@ public void sortApplications(Comparator<Application> c, String orderBy) {
     public void updateFilteredApplicationList(Predicate<Application> predicate) {
         requireNonNull(predicate);
         filteredApplications.setPredicate(predicate);
+        this.currentFilter = predicate;
+    }
+
+    @Override
+    public void updateFilteredApplicationList() {
+        updateFilteredApplicationList(this.currentFilter);
```

Comment:
Nice job implementing this! Changes minimal code whilst giving the desired behavior.